src/Azure.Functions.Cli/Actions/DeployActions/Platforms/KnativePlatform.cs

```diff
@@ -0,0 +1,147 @@
+        {
+            await Deploy(functionName, image, FUNCTIONS_NAMESPACE, min, max);
+        }
+
+        public KnativePlatform(string configFile)
+        {
+            this.configFile = configFile;
+            KubeClientOptions options;
+
+            if (!string.IsNullOrEmpty(configFile))
+            {
+                options = K8sConfig.Load(configFile).ToKubeClientOptions(defaultKubeNamespace: FUNCTIONS_NAMESPACE);
+            }
+            else
+            {
+                options = K8sConfig.Load().ToKubeClientOptions(defaultKubeNamespace: FUNCTIONS_NAMESPACE);
+            }
+
+            client = KubeApiClient.Create(options);
+        }
+
+        private async Task Deploy(string name, string image, string nameSpace, int min, int max)
+        {
+            var isHTTP = IsHTTPTrigger(name);
+
+            await CreateNamespace(nameSpace);
+            client.DefaultNamespace = nameSpace;
+
+            ColoredConsole.WriteLine();
+            ColoredConsole.WriteLine("Deploying function to Knative...");
+
+            var knativeService = GetKnativeService(name, image, nameSpace, min, max, isHTTP);
+            var json = Newtonsoft.Json.JsonConvert.SerializeObject(knativeService,
+                            Newtonsoft.Json.Formatting.None,
+                            new Newtonsoft.Json.JsonSerializerSettings
+                            {
+                                NullValueHandling = Newtonsoft.Json.NullValueHandling.Ignore
+                            });
+
+            File.WriteAllText("deployment.json", json);
+            await KubernetesHelper.RunKubectl($"apply -f deployment.json");
+            File.Delete("deployment.json");
+
+            var externalIP = await GetIstioClusterIngressIP();
+            if (externalIP == "")
+            {
+                ColoredConsole.WriteLine("Couldn't find Istio Cluster Ingress External IP");
+                return;
+            }
+
+            var host = GetFunctionHost(name, nameSpace);
+
+            ColoredConsole.WriteLine("");
+
+            ColoredConsole.WriteLine("Function deployed successfully!");
+            ColoredConsole.WriteLine();
+            ColoredConsole.WriteLine($"Function URL: http://{externalIP}");
+            ColoredConsole.WriteLine($"Function Host: {host}");
+            ColoredConsole.WriteLine("");
+            ColoredConsole.WriteLine("Plese note: it may take a few minutes for the knative service to be reachable");
+        }
+
+        private string GetFunctionHost(string functionName, string nameSpace) {
+            return string.Format("{0}.{1}.example.com", functionName, nameSpace);
+        }
+
+        private bool IsHTTPTrigger(string functionName)
+        {
+            var str = File.ReadAllText(string.Format("{0}/function.json", functionName));
+            var jObj = JsonConvert.DeserializeObject<FunctionJson>(str);
+            return jObj.bindings.Any(d=> d.type == "http");
```

Comment:
I think you want `httpTrigger`, right?

Reply:
I was looking at the out binding, you're correct the in binding is what should be looked at